src/responder/common/cache_req/cache_req_domain.c

```diff
@@ -155,6 +161,14 @@ cache_req_domain_new_list_from_string_list(TALLOC_CTX *mem_ctx,
         cr_domain->fqnames =
             cache_req_domain_use_fqnames(dom, enforce_non_fqnames);
 
+        /* when using the domain resolution order, using shortnames as input
+         * is allowed by default. However, we really want to use the fully
+         * qualified name as output in order to avoid conflicts whith users
+         * who have the very same name. */
+        if (resolution_order != NULL) {
+            cr_domain->domain->output_fqnames = true;
```

Comment:
I'm sorry I didn't notice this the first time I was reviewing the patch, but I don't like the cache_req request code modifying the domain property. Could we please do this instead at the time the resolution order is read?

Additionally, I wonder if we could have a setter for the output_fqnames property. We've been wanting to make the `sss_domain_info` opaque precisely to control who can set what.